Userland/Applications/PackageManager/InstalledPort.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Function.h>
+#include <Applications/PackageManager/InstalledPort.h>
+#include <LibCore/File.h>
+#include <LibCore/System.h>
+
+ErrorOr<Vector<InstalledPort>> InstalledPort::read_ports_database()
+{
+    auto input_stream = TRY(Core::File::open("/usr/Ports/installed.db"sv, Core::File::OpenMode::Read));
+    auto whole_file = TRY(input_stream->read_until_eof());
```

Comment:
We might have discussed this already, I don't remember, but could we use `BufferedFile::read_line()` here?

Reply:
Sure, this will be better! I didn't know about this piece of code.